pootle/apps/pootle_fs/matcher.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import logging
+import os
+from fnmatch import fnmatch
+
+
+logger = logging.getLogger(__name__)
+
+
+class FSPathMatcher(object):
+
+    def __init__(self, context):
+        self.context = context
+
+    @property
+    def lang_mapper(self):
+        return self.context.lang_mapper
+
+    @property
+    def project(self):
+        return self.context.project
+
+    @property
+    def root_directory(self):
+        return self.context.project.local_fs_path
+
+    @property
+    def translation_path(self):
+        return self.context.project.config[
+            "pootle_fs.translation_paths"]["default"]
+
+    def make_pootle_path(self, **matched):
+        language_code = matched.get("language_code")
+        filename = matched.get("filename")
+        ext = matched.get("ext")
+        if not (language_code and filename and ext):
+            return
+        return "/".join(
+            ["", language_code,
+             self.project.code]
+            + [m for m in
+               matched.get('dir_path', '').split("/")
+               if m]
+            + ["%s.%s"
+               % (filename, ext)])
+
+    def match_pootle_path(self, pootle_path_match=None, **matched):
+        pootle_path = self.make_pootle_path(**matched)
```

Comment:
its for reverse matching - ie given a set of lang/subdirs/filename would our `translation_path` give us a fs path, given the fs/pootle glob filter